lib/archethic/transaction_chain/transaction/validation_stamp/ledger_operations.ex

```diff
@@ -157,31 +195,31 @@ defmodule Archethic.TransactionChain.Transaction.ValidationStamp.LedgerOperation
       ...> }
       ...> |> LedgerOperations.sufficient_funds?([
       ...>     %UnspentOutput{from: "@Charlie5", amount: 3_000_000_000, type: :UCO},
-      ...>     %UnspentOutput{from: "@Bob4", amount: 10_000_000_000, type: {:NFT, "@BobNFT"}}
+      ...>     %UnspentOutput{from: "@Bob4", amount: 100_000_000, type: {:NFT, "@BobNFT", 10}}
       ...> ])
       true
   """
   @spec sufficient_funds?(t(), list(UnspentOutput.t() | TransactionInput.t())) :: boolean()
   def sufficient_funds?(operations = %__MODULE__{}, inputs) when is_list(inputs) do
-    %{uco: uco_balance, nft: nfts_received} = ledger_balances(inputs)
+    %{uco: uco_balance, nft: nfts_received} = utxo_ledger_balances(inputs)
     %{uco: uco_to_spend, nft: nfts_to_spend} = total_to_spend(operations)
-    uco_balance >= uco_to_spend and sufficient_nfts?(nfts_received, nfts_to_spend)
+    uco_balance >= uco_to_spend and nfts_exist?(nfts_received, nfts_to_spend)
   end
 
-  defp sufficient_nfts?(nfts_received = %{}, nft_to_spend = %{})
+  defp nfts_exist?(nfts_received = %{}, nft_to_spend = %{})
        when map_size(nfts_received) == 0 and map_size(nft_to_spend) > 0,
        do: false
 
-  defp sufficient_nfts?(_nfts_received, nfts_to_spend) when map_size(nfts_to_spend) == 0, do: true
+  defp nfts_exist?(_nfts_received, nfts_to_spend) when map_size(nfts_to_spend) == 0, do: true
 
-  defp sufficient_nfts?(nfts_received, nfts_to_spend) do
-    Enum.all?(nfts_to_spend, fn {nft_address, amount_to_spend} ->
+  defp nfts_exist?(nfts_received, nfts_to_spend) do
+    Enum.all?(nfts_to_spend, fn {nft_address, {_to, to_nft_id}} ->
```

Comment:
you are not handling balance, but this function is to check if you have enough NFT utxo to spend